tmt/steps/prepare/artifact/providers/repository_provider.py

```diff
@@ -0,0 +1,108 @@
+"""
+Artifact providers for Repository and Repository Files.
+"""
+
+from collections.abc import Iterator
+from shlex import quote
+from urllib.parse import urlparse
+
+import tmt.log
+import tmt.utils
+from tmt.container import container
+from tmt.steps.prepare.artifact.providers import (
+    ArtifactProvider,
+    DownloadError,
+)
+from tmt.steps.prepare.artifact.providers.info import ArtifactInfo
+from tmt.steps.provision import Guest
+from tmt.utils import GeneralError, Path, ShellScript
+
+
+@container
+class RepositoryFileInfo(ArtifactInfo):
+    """
+    A single repository file channel.
+
+    Encapsulates information about a repository file, including its URL and
+    derived identifier (filename). It validates that the URL uses HTTP/HTTPS
+    and points to a file with a ``.repo`` extension.
+    """
+
+    _raw_artifact: str
+
+    def __post_init__(self) -> None:
+        parsed_url = urlparse(self._raw_artifact)
+        if parsed_url.scheme not in ('http', 'https'):
+            raise ValueError(
+                f"Invalid repository URL: {self._raw_artifact}. Only HTTP(S) is supported."
+            )
+        path = tmt.utils.Path(parsed_url.path)
+        if path.suffix != '.repo':
```

Comment:
I am not sure we need to be here so strict, what if user has a URL with query params ...

```
pathlib.Path("https://abc.com/what.repo?wtf").suffix == ".repo?wtf"
```